### ISSUE TITLE
Create `new` methods with Config arguments for existing builders

### ISSUE THREAD

**Issue by Rjected** [paradigmxyz/reth#5835]
### Describe the feature

ref https://github.com/paradigmxyz/reth/pull/5834#discussion_r1432934073

We currently have `From` impls in the `reth-config` crate:
https://github.com/paradigmxyz/reth/blob/d673f5fbfbe3bf9e21d1b0d147d9facc39a580d2/crates/config/src/config.rs#L179-L181

Instead, we could create a method with this signature and remove the `From` impl:
```rust
    /// Creates a new [BodiesDownloaderBuilder] from the given [BodiesConfig].
    pub fn new(bodies_config: BodiesConfig) -> Self {
```

It would also make sense to come up with an easy to use method to replace `Config::network_config`

**Comment by yash-atreya** [comment]
Can take this

**Comment by Rjected** [comment]
> Can take this

most of this issue was solved with #5837, feel free to submit a PR for this part:
> It would also make sense to come up with an easy to use method to replace Config::network_config

**Comment by yash-atreya** [comment]
> It would also make sense to come up with an easy-to-use method to replace `Config::network_config`

Hey @Rjected, sorry for the delay. Can you please elaborate more on what needs to be done as a `new` method already exists for `NetworkConfigBuilder`. The only difference is ingesting the peers file and discovery.
